components/combobox/src/auro-combobox.js

```diff
@@ -915,7 +915,16 @@ export class AuroCombobox extends AuroElement {
    */
   reset() {
     this.input.reset();
+    this.menu.value = undefined;
```

Comment:
This line isn't needed. I just tested without it and the menu correctly resets it's value to undefined when the input is reset.